ts/src/vector/filter/sequenceSelectionVector.spec.ts

```diff
@@ -0,0 +1,202 @@
+            seqVector.setLimit(3);
+            expect(seqVector.limit).toBe(3);
+            expect(seqVector.capacity).toBe(5);
+        });
+    });
+
+    describe("selectionValues Test", () => {
+        it("Should return selectionVector", () => {
+            const values = seqVector.selectionValues();
+            expect(values).toStrictEqual([0, 1, 2, 3, 4]);
+        });
+
+        it("Should return empty array for empty vector", () => {
+            const emptyVector = new SequenceSelectionVector(0, 1, 0);
+            expect(emptyVector.selectionValues()).toStrictEqual([]);
+        });
+
+        it("Should return correct values with custom base and delta", () => {
+            const customVector = new SequenceSelectionVector(10, 5, 5);
+            expect(customVector.selectionValues()).toStrictEqual([10, 15, 20, 25, 30]);
+        });
+
+        it("Should return correct values with negative delta", () => {
+            const negVector = new SequenceSelectionVector(100, -10, 5);
+            expect(negVector.selectionValues()).toStrictEqual([100, 90, 80, 70, 60]);
+        });
+
+        it("Should cache materialized array", () => {
+            const values1 = seqVector.selectionValues();
+            const values2 = seqVector.selectionValues();
+            expect(values1).toBe(values2); // Same reference
+        });
+
+        it("Should return materialized array after setIndex", () => {
+            seqVector.setIndex(2, 999);
+            const values = seqVector.selectionValues();
+            expect(values).toStrictEqual([0, 1, 999, 3, 4]);
+        });
+    });
+
+    describe("get capacity Test", () => {
+        it("Should return capacity", () => {
+            expect(seqVector.capacity).toBe(5);
+        });
+
+        it("Should return 0 for empty vector", () => {
+            const emptyVector = new SequenceSelectionVector(0, 1, 0);
+            expect(emptyVector.capacity).toBe(0);
+        });
+
+        it("Should return correct capacity for large vector", () => {
+            const largeVector = new SequenceSelectionVector(0, 1, 1000);
+            expect(largeVector.capacity).toBe(1000);
+        });
+    });
+
+    describe("get limit Test", () => {
+        it("Should return limit", () => {
+            expect(seqVector.limit).toBe(5);
+        });
+
+        it("Should return 0 for empty vector", () => {
+            const emptyVector = new SequenceSelectionVector(0, 1, 0);
+            expect(emptyVector.limit).toBe(0);
+        });
+
+        it("Should return updated limit after setLimit", () => {
+            seqVector.setLimit(3);
+            expect(seqVector.limit).toBe(3);
+        });
+
+        it("Should initialize limit to size", () => {
+            const customVector = new SequenceSelectionVector(10, 5, 50);
+            expect(customVector.limit).toBe(50);
+            expect(customVector.capacity).toBe(50);
+        });
+    });
+
+    describe("Memory efficiency Test", () => {
```

Comment:
I think you'll have a hard time actually testing this.
Or to put it in other words, if the internal implementation would change these tests would still pass, so I don't think they are adding valuable information in terms of testing.
Consider removing them.